### PR TITLE
get embedding for a single file

### PR DESCRIPTION
**Pull request by nithinraok**
Signed-off-by: nithinraok <nithinrao.koluguri@gmail.com>

Usage:
`speaker_model = ExtractSpeakerEmbeddingsModel.from_pretrained(model_name="speakerverification_speakernet")`
` embs = speaker_model.get_embedding('audio_path')`

### CODE REVIEW

File: nemo/collections/asr/models/label_models.py [hunk 1]
```diff
@@ -337,6 +338,23 @@ def setup_finetune_model(self, model_config: DictConfig):
 
         logging.info(f"Changed decoder output to # {self.decoder._num_classes} classes.")
 
+    @torch.no_grad()
+    def get_embedding(self, path2audio_file):
+        audio, sr = librosa.load(path2audio_file, sr=None)
+        if sr != 16000:
+            raise ValueError("Make sure input audio has sample rate of 16Khz")
```
Comment (jbalam-nv) [inline nemo/collections/asr/models/label_models.py:345]: Why are we throwing an error here? Shouldn't we just resample the audio to 16 kHz? That is the current behavior in ASR.

Comment (nithinraok) [reply nemo/collections/asr/models/label_models.py:345]: we could do that as well

Comment (titu1994) [reply nemo/collections/asr/models/label_models.py:345]: Resampling it is better.

**Review by titu1994 — COMMENTED** [review]
If you don't need gradients, might as well explicitly put model.eval() inside this function. Also make sure that all tensors that are moved to gpu are explicitly deleted at the end of the function.

File: nemo/collections/asr/models/label_models.py [hunk 2]
```diff
@@ -337,6 +338,23 @@ def setup_finetune_model(self, model_config: DictConfig):
 
         logging.info(f"Changed decoder output to # {self.decoder._num_classes} classes.")
 
+    @torch.no_grad()
+    def get_embedding(self, path2audio_file):
+        audio, sr = librosa.load(path2audio_file, sr=None)
+        if sr != 16000:
+            audio = librosa.core.resample(audio, sr, 16000)
```
Comment (jbalam-nv) [inline nemo/collections/asr/models/label_models.py:345]: The target sampling rate should be read from the model config. This will fail for a model that was trained with say 8 kHz data.

Comment (nithinraok) [reply nemo/collections/asr/models/label_models.py:345]: I guess the whole idea is to not read from config, hence the quick way to get embeddings for a single audio file and thought about providing the target sample rate, but since all speaker models are 16Khz hence the 16000 number.

Comment (titu1994) [reply nemo/collections/asr/models/label_models.py:345]: No that doesn't mean the default should be 16khz. The models SR should be used, whatever it is, from it's own config. You can use self.cfg to access the config that was used to build the model c

**Review by titu1994 — APPROVED** [review]
Looks good.
